Remotion/Core/Core/Context/SafeContextBoundary.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace Remotion.Context
+{
+  /// <summary>
+  /// Represents a <see cref="SafeContext"/> boundary, across which values will not flow.
+  /// The previous context is restored when disposed.
+  /// </summary>
+  /// <seealso cref="ISafeContextStorageProvider.OpenSafeContextBoundary"/>
+  public readonly struct SafeContextBoundary : IDisposable
+  {
+    private readonly ISafeContextBoundaryStrategy _boundaryStrategy;
```

Comment:
Why isn't NRT complaining that the `?` is missing?

Reply:
Struct values could always be `null`, but that would not make sense with NRT. So, it is possible to define values that are non-nullable and the null checks guard only the assignment part in the constructor.